app/src/layouts/DashboardLayout/Sidebar/PrimarySidebar/components/JoinSlackButton/JoinSlackButton.tsx

```diff
@@ -0,0 +1,41 @@
+import { RQButton } from "lib/design-system/components";
+import SlackIcon from "assets/icons/slack.svg?react";
+import sendSlackInvite from "components/misc/SupportPanel/sendSlackInvite";
+import React, { useState } from "react";
+import starAnimation from "assets/images/gifs/Stars.gif";
+import "./joinSlackButton.scss";
+import { trackSlackConnectClicked } from "modules/analytics/events/misc/UnifiedSupport";
+
+const handleJoinSlack = async (setDisabled: React.Dispatch<React.SetStateAction<boolean>>) => {
```

Comment:
instead of this complicated type definition, wrap the function in `useCallback` and use `setDisabled` without passing the arg.

L#27 then becomes `onclick={handleJoinSlack}` which is also much cleaner